### PR TITLE
Bugfix/email production fix

### LINKED ISSUE #162: Confirmation email uses port 3001 instead of port 3000
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

### LINKED ISSUE #166: Emailing not working on production
**Describe the bug**
Currently get a server error when attempting to email confirmations

**To Reproduce**
1. Go to fortunacombat.com
2. Click on 'Signup'
3. Enter information and press Signup
4. See 'Server Error' output on the UI

**Expected behavior**
The email sent prompt should be coming out.

**Screenshots**

**Additional context**
For what it's worth, the account still gets created in the DB, so it's fine on that part, but I should get on fixing the mailing to clients asap.

### PR DESCRIPTION
**Pull request by Derplime**
**Description**
Changed our SMTP service for emails to SendGrid from Gmail.
Added some error handling to some of the email related routes in the userController.
Email link now works correctly for both development and production use.

With this, I updated the `.env` file in the drive with the one that master will use when it is merged with this branch. The `.env` file on the server is already up to date with this version.

Resolves #166 
Resolves #162 

**Type of change**
Check options that are relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Fix or feature that would cause existing functionality to not work as expected
- [x] This change requires a update in the design document

**Steps to Verify Functionality**
- Download new `.env` file from Drive and put into `Fortuna/` directory
- Signup New User (locally)
- Use email on confirmation
- Confirm the user is confirmed via frontend/DB

**Final Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Pulled updated master branch (if changed since branch creation) and corrected any conflicts, if any occur.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject",
     "flow": "flow",
-    "devserver": "nodemon ./lib/backend/server.js"
+    "devserver": "NODE_ENV=development nodemon ./lib/backend/server.js"
```
Comment (SecondThread) [inline package.json:35]: um, if you say so

File: src/backend/controllers/userController.js [hunk 2]
```diff
@@ -21,6 +21,8 @@ const Token = require('../../models/tokenModel');
 const regeneratorRuntime = require("regenerator-runtime");
 // JWT Secret
 const jwtSecret = process.env.JWT_SECRET;
+// Front-End Host Constant
+const FRONTEND = (process.env.NODE_ENV === 'development') ? 'localhost:3000' : process.env.FRONTEND_HOST;
```
Comment (SecondThread) [inline src/backend/controllers/userController.js:25]: cool

**Review by SecondThread — APPROVED** [review]
looks good to me, thanks for the error handling. (when it just dies silently it is pretty annoying)

File: package.json (hunk 1 above)
Comment (Derplime) [reply package.json:35]: To clarify on that I don't do `npm run devserver` with the pm2 process, I just do `node ./lib/backend/server.js` for that

devserver is just when we do our stuff locally.

**Pull request merged**